packages/@aws-cdk/aws-logs/lib/log-retention.ts

```diff
@@ -0,0 +1,167 @@
+   * @default - A new role is created
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Retry options for all AWS API calls.
+   *
+   * @default - AWS SDK default retry options
+   */
+  readonly logRetentionRetryOptions?: LogRetentionRetryOptions;
+}
+
+/**
+ * Retry options for all AWS API calls.
+ */
+export interface LogRetentionRetryOptions {
+  /**
+   * The maximum amount of retries.
+   *
+   * @default 3 (AWS SDK default)
+   */
+  readonly maxRetries?: number;
+  /**
+   * The base duration to use in the exponential backoff for operation retries.
+   *
+   * @default Duration.millis(100) (AWS SDK default)
+   */
+  readonly base?: cdk.Duration;
+}
+
+/**
+ * Creates a custom resource to control the retention policy of a CloudWatch Logs
+ * log group. The log group is created if it doesn't already exist. The policy
+ * is removed when `retentionDays` is `undefined` or equal to `Infinity`.
+ */
+export class LogRetention extends cdk.Construct {
+
+  /**
+   * The ARN of the LogGroup.
+   */
+  public readonly logGroupArn: string;
+
+  constructor(scope: cdk.Construct, id: string, props: LogRetentionProps) {
+    super(scope, id);
+
+    // Custom resource provider
+    const provider = this.ensureSingletonLogRetentionFunction(props);
+
+    // Need to use a CfnResource here to prevent lerna dependency cycles
+    // @aws-cdk/aws-cloudformation -> @aws-cdk/aws-lambda -> @aws-cdk/aws-cloudformation
+    const retryOptions = props.logRetentionRetryOptions;
+    const resource = new cdk.CfnResource(this, 'Resource', {
+      type: 'Custom::LogRetention',
+      properties: {
+        ServiceToken: provider.functionArn,
+        LogGroupName: props.logGroupName,
+        SdkRetry: retryOptions ? {
+          maxRetries: retryOptions.maxRetries,
+          base: retryOptions.base?.toMilliseconds(),
+        } : undefined,
+        RetentionInDays: props.retention === RetentionDays.INFINITE ? undefined : props.retention,
+      },
+    });
+
+    const logGroupName = resource.getAtt('LogGroupName').toString();
+    // Append ':*' at the end of the ARN to match with how CloudFormation does this for LogGroup ARNs
+    // See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-loggroup.html#aws-resource-logs-loggroup-return-values
+    this.logGroupArn = cdk.Stack.of(this).formatArn({
+      service: 'logs',
+      resource: 'log-group',
+      resourceName: `${logGroupName}:*`,
+      sep: ':',
+    });
+  }
+
+  /**
+   * Helper method to ensure that only one instance of LogRetentionFunction resources are in the stack mimicking the
+   * behaviour of @aws-cdk/aws-lambda's SingletonFunction to prevent circular dependencies
+   */
+  private ensureSingletonLogRetentionFunction(props: LogRetentionProps) {
+    const functionLogicalId = 'LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8a';
+    const existing = cdk.Stack.of(this).node.tryFindChild(functionLogicalId);
+    if (existing) {
+      // Just assume this is true
```

Comment:
Delete this comment. Not clear what this means.

```suggestion
```

Reply:
the effect of copy/pasting existing code